hub/core/compute/thread.py

```diff
@@ -9,3 +9,8 @@ def __init__(self, workers):
 
     def map(self, func, iterable):
         return self.pool.map(func, iterable)
+
+    def close(self):
```

Comment:
all of these compute contributions are out of context, is this a problem?

Reply:
mac tests were failing, because of compute providers not being closed, made hot fixes to multiple PRs to fix this